src/content/structured/components/side-nav/code.mdx

```diff
@@ -744,3 +749,570 @@ export const withReactRouter = [
+      >
+        <path
+          d="M12 6.19L17 10.69V18.5H15V12.5H9V18.5H7V10.69L12 6.19ZM12 3.5L2 12.5H5V20.5H11V14.5H13V20.5H19V12.5H22L12 3.5Z"
+          fill="currentColor"
+        />
+      </svg>
+    </IcNavigationItem>
+    <IcNavigationItem
+      slot="secondary-navigation"
+      href="/"
+      label="Settings"
+    >
+      <svg
+        slot="icon"
+        width="24"
+        height="24"
+        viewBox="0 0 24 24"
+        fill="none"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <path
+          d="M12 6.19L17 10.69V18.5H15V12.5H9V18.5H7V10.69L12 6.19ZM12 3.5L2 12.5H5V20.5H11V14.5H13V20.5H19V12.5H22L12 3.5Z"
+          fill="currentColor"
+        />
+      </svg>
+    </IcNavigationItem>
+  </IcSideNavigation>
+  <div style={{display: 'flex', flexDirection: 'column', flexGrow: '1'}}>
+    <main style={{height: '100vh'}}>
+      <IcSectionContainer>
+        <IcTypography>
+          <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed vestibulum venenatis facilisis. Nam tortor felis, auctor vel ante quis, tempor interdum libero. In dictum sodales velit, eu egestas arcu dignissim ac. Aliquam facilisis eros dolor, id laoreet orci sagittis ut. Sed tempus, lacus in pretium molestie, lectus magna interdum risus, vel fringilla libero ex eu dui. Suspendisse ullamcorper vehicula lacinia. Phasellus congue velit nisl, vitae congue ligula rutrum id.</p>
+        </IcTypography>
+        <IcTypography>
+          <p>Etiam in suscipit metus. Duis semper, sapien a molestie semper, ex nibh porttitor tellus, vel molestie justo odio vel purus. Curabitur porttitor, tortor sed semper sollicitudin, odio odio congue tortor, eget pulvinar tellus nisl ac lacus. In ultricies commodo lorem, a laoreet diam. Ut a mauris at tellus tincidunt ullamcorper sit amet in metus. Aenean facilisis placerat dictum. Phasellus mattis ante sollicitudin luctus iaculis. Nam porttitor lobortis rhoncus. Nam nec malesuada purus, at pulvinar mauris. Nam non lorem ante.</p>
+        </IcTypography>
+      </IcSectionContainer>
+    </main>
+    <IcFooter
+      description="The ICDS is maintained by the Design Practice Team. If you've got a question or want to feedback,
+        please get in touch."
+      caption="All content is available under the Open Government Licence v3.0, except source code and code examples which are available under the MIT Licence."
+    >
+      <IcFooterLink slot="link" href="/">Get Started</IcFooterLink>
+      <IcFooterLink slot="link" href="/">Accessibility</IcFooterLink>
+      <IcFooterLink slot="link" href="/">Styles</IcFooterLink>
+    </IcFooter>
+    <IcClassificationBanner classification="official"></IcClassificationBanner>
+  </div>
+</div>`,
+  },
+];
+
+<ComponentPreview
+  snippets={basicLayoutFixedViewport}
+  style={{
+    display: "flex",
+    height: "100%",
+    justifyContent: "flex-start",
+    overflow: "hidden",
+    position: "relative",
+    padding: "0px",
+  }}
+/>
```

Comment:
Do we need an actual preview of the component adding here? At the moment it just shows the code snippet

Reply:
As it's an example which involves the side navigation being fixed to the viewport, this would interfere with the rest of the page instead of being incapsulated within the component preview. I can take another look and see if I can use an iframe from a static HTML page and pass that through into the component preview.

Reply:
Ah ok sounds good :) If we can't manage to get it working however then it might be worth adding a line of text like 'This variant will be fixed to the viewport' or something? So people know what it would look like / how it would appear if they can't see a preview

Reply:
https://github.com/mi6/ic-design-system/pull/163#issuecomment-1376008666